extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -129,18 +136,41 @@ class AmpStickyAd extends AMP.BaseElement {
         const borderBottom = this.element./*OK*/offsetHeight;
         this.viewport_.updatePaddingBottom(borderBottom);
         this.addCloseButton_();
-        timer.delay(() => {
-          // Unfortunately we don't really have a good way to measure how long it
-          // takes to load an ad, so we'll just pretend it takes 1 second for
-          // now.
-          this.vsync_.mutate(() => {
-            this.element.classList.add('amp-sticky-ad-loaded');
-          });
-        }, 1000);
+      });
+      // Get Ad promise for ad render-start
+      this.adRenderStartPromise_ = this.ad_.implementation_.renderStartPromise;
+      if (this.adRenderStartPromise_) {
+        this.delayAdLoad_();
+      } else {
+        //add listener for element not attached and not built
+        this.ad_.addEventListener('amp:built', () => {
+          this.adRenderStartPromise_ =
+              this.ad_.implementation_.renderStartPromise;
+          this.delayAdLoad_();
+          // schedule Layout again after amp-ad is built.
+          this.scheduleLayout(this.ad_);
+        });
+      }
+    }
+  }
+
+  delayAdLoad_() {
+    const type = this.ad_.getAttribute('type');
```

Comment:
Not all, only those that we determine should support `render-start`, right? Generally, blocking `layoutCallback` will eventually be the right thing to do. But if you feel like we're not ready to do so, we can support `render-start` the same way - via events. Then, our code will look like:

```
  this.ad_.addEventListener('amp:render:start', () => {
    // TODO: finish sticky ad set up.
  });
```

This might be a simple thing as well for now.